tide/core/ScreenLock.h

```diff
@@ -0,0 +1,131 @@
+/* conditions are met:                                               */
+/*                                                                   */
+/*   1. Redistributions of source code must retain the above         */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer.                                                  */
+/*                                                                   */
+/*   2. Redistributions in binary form must reproduce the above      */
+/*      copyright notice, this list of conditions and the following  */
+/*      disclaimer in the documentation and/or other materials       */
+/*      provided with the distribution.                              */
+/*                                                                   */
+/*    THIS  SOFTWARE IS PROVIDED  BY THE  UNIVERSITY OF  TEXAS AT    */
+/*    AUSTIN  ``AS IS''  AND ANY  EXPRESS OR  IMPLIED WARRANTIES,    */
+/*    INCLUDING, BUT  NOT LIMITED  TO, THE IMPLIED  WARRANTIES OF    */
+/*    MERCHANTABILITY  AND FITNESS FOR  A PARTICULAR  PURPOSE ARE    */
+/*    DISCLAIMED.  IN  NO EVENT SHALL THE UNIVERSITY  OF TEXAS AT    */
+/*    AUSTIN OR CONTRIBUTORS BE  LIABLE FOR ANY DIRECT, INDIRECT,    */
+/*    INCIDENTAL,  SPECIAL, EXEMPLARY,  OR  CONSEQUENTIAL DAMAGES    */
+/*    (INCLUDING, BUT  NOT LIMITED TO,  PROCUREMENT OF SUBSTITUTE    */
+/*    GOODS  OR  SERVICES; LOSS  OF  USE,  DATA,  OR PROFITS;  OR    */
+/*    BUSINESS INTERRUPTION) HOWEVER CAUSED  AND ON ANY THEORY OF    */
+/*    LIABILITY, WHETHER  IN CONTRACT, STRICT  LIABILITY, OR TORT    */
+/*    (INCLUDING NEGLIGENCE OR OTHERWISE)  ARISING IN ANY WAY OUT    */
+/*    OF  THE  USE OF  THIS  SOFTWARE,  EVEN  IF ADVISED  OF  THE    */
+/*    POSSIBILITY OF SUCH DAMAGE.                                    */
+/*                                                                   */
+/* The views and conclusions contained in the software and           */
+/* documentation are those of the authors and should not be          */
+/* interpreted as representing official policies, either expressed   */
+/* or implied, of Ecole polytechnique federale de Lausanne.          */
+/*********************************************************************/
+
+#ifndef SCREENLOCK_H
+#define SCREENLOCK_H
+
+#include "serialization/includes.h"
+#include "types.h"
+
+#include <QObject>
+#include <boost/enable_shared_from_this.hpp>
+
+/**
+ * Allow wall lock preventing unwanted streams from opening and making Web
+ * Interface view-only.
+ */
+class ScreenLock : public QObject,
+                   public boost::enable_shared_from_this<ScreenLock>
+{
+    Q_OBJECT
+
+    Q_PROPERTY(bool locked READ isLocked NOTIFY lockChanged)
+    Q_PROPERTY(
+        QStringList streamList READ getStreamList NOTIFY streamListChanged)
+
+public:
+    /** Constructor */
+    ScreenLock();
+
+    /** Close the specified stream. */
+    Q_INVOKABLE void closeStream(const QString uri);
```

Comment:
in closeStream method I emit streamClosed(uri), to close a hidden stream, which is not needed in deregisterStream as the stream is closed elsewhere.